journal.py

```diff
@@ -226,23 +228,16 @@ def editview_entry(request):
 @view_config(route_name='add', renderer='json', request_method="POST")
 def add_entry(request):
     if request.authenticated_userid:
-        try:
-            write_entry(request)
-        except psycopg2.Error:
-            # this will catch any errors generated by the database
-            return HTTPInternalServerError()
-
-        cursor = request.db.cursor()
-        cursor.execute(NEW_ENTRY)
-        keys = ('id', 'title', 'text', 'created')
-        row = cursor.fetchone()
-        entry = dict(zip(keys, row))
-        entry['text'] = markdown.markdown(entry['text'], extensions=['codehilite', 'fenced_code'])
-        entry['created'] = entry['created'].strftime('%b.%d.%Y')
-        return entry
+        if request.method == 'POST':
```

Comment:
It's a bit redundant to check if the request method is POST here, since you have a predicate in your view config that asserts that it must be.  Pyramid won't allow any GET requests to reach this view method.